profile-specific-challenges/frontend/components/idioms.jsx

```diff
@@ -7,36 +7,51 @@
   Some references used:
   https://claritydev.net/blog/the-most-common-mistakes-when-using-react
 */
-import { useEffect } from "react";
+import { useCallback, useContext, useEffect } from "react";
 
+// This was missing a return but that showButton could be abstracted out.
```

Comment:
Sorry for the sloppyiness on this file - that's not the issue with the function. 

It is a bad pattern to make a function that renders jsx without being a component. This is redefining a component on every render, plus does not follow the component convention that it should be named uppercase.